File: q30-q39/q35/q35.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        left = 0
        right = len(nums) - 1
        middle = 0 
        while left < right:
            middle = (left + right) // 2
            if nums[middle] == target:
                return middle
            if nums[middle] <= target:
                left = middle + 1
            else:
                right = middle - 1
        middle = int ((left + right) / 2)
        print (middle)
        print(left)
        print(right)
        if nums[middle] < target:
            return middle + 1
        return middle

File: q30-q39/q35/q35.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int left = 0, right = nums.size() -1;
        int middle = 0;
        while (left < right){
            middle = (left + right) / 2;
            if (nums[middle] == target){
                return middle;
            }
            if (nums[middle] <= target)
                left = middle + 1;
            else
                right = middle - 1;
        }
        middle = (left + right) / 2;
        if (target > nums[middle])
            return middle + 1;
        else 
            return middle;
    }
};
